### DISCUSSION TITLE
Photoprism portable

### DISCUSSION THREAD

**Discussion by nautatva** [photoprism/photoprism#1893] [Ideas]
I have multiple PCs and I don't want to dedicate one specifically to this purpose. Is it possible to just connect a hard disk to any PC and start photoprism?

I don't mind having caches/data on hard disk itself as long as the task is fulfilled. How should I set this up?

**Comment by lastzero** [comment]
Theoretically, yes, but that requires experience with Docker. Alternatively, setting up a custom runtime environment requires even more experience. If we recommended that, we would also have to provide support to anyone who tried it, which would leave no time for development.

**Reply by lastzero** [reply]
A simple solution could be to install PhotoPrism in a VirtualBox VM so you can start it on most common operating systems. Running separate environments on each computer is a lot of work or requires management infrastructure to reduce manual work.

**Reply by nautatva** [reply]
Unfortunately I don't have any experience with docker, but I do know that docker is kind of a minimal VM in itself. Can I leverage that? Perhaps I can install on hard disk, and keep all the data on disk itself. Any temporary directories/anything I am missing here?

**Reply by nautatva** [reply]
Need to keep the same path on all the PCs is one requirement I see. That can be maintained. Any problems with this approach?

**Reply by lastzero** [reply]
Docker uses Linux Containers which essentially are namespaces (thus the containerized app can't see all folders), so not a portable VM like VirtualBox which runs a complete operating system with own kernel etc.

**Comment by centralhardware** [comment]
it's not goal of photoprism project but server software suggest to ensure fault tolerance of the storage system. On server it is raid even for RPI. and use single hdd for store data you will risk data being lost for a million different reasons you can't do anything about.
